hashintel/hash
# H-1331: add ability to count the incoming/outgoing links of an entity in a bar graph

## 🌟 What is the purpose of this PR?




This PR adds a "Count links" variant to the bar graph block, counting the incoming or outgoing links of a configurable link entity type of entities of a certain type.

## 🔗 Related links





- H-1331

## 🚫 Blocked by




- [x] merging https://github.com/hashintel/hash/pull/3658

## 🔍 What does this change?




See commits.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



- Currently the `getEntityIncomingLinks` and `getEntityOutgoingLinks` non-temporal methods break when they are passed a temporal subgraph (H-1522)

## 🐾 Next steps



## 🛡 What tests cover this?



Manual testing.

## ❓ How to test this?



1. Checkout the branch locally and run the block in development mode
2. Load the locally hosted block into a page, and add a data query which selects entities that have either incoming or outgoing links
3. Configure the "Count Links" bar chart variant and select the relevant linked entity type you'd like to count in the bar graph
4. Click "Update Chart", and observe the number of outgoing or incoming links being displayed in the chart

## 📹 Demo




https://github.com/hashintel/hash/assets/42802102/9b4788e3-2326-416d-b67d-96d6624f4a7b
